src/services/walletProviders/keyBasedWallet.js

```diff
@@ -173,7 +173,7 @@ export default class KeyBasedWalletProvider {
     return Promise.resolve({ nonce, transactionCount });
   }
 
-  async getTransactionCount(walletAddress: string) {
+  getTransactionCount(walletAddress: string) {
     return this.wallet.provider.getTransactionCount(walletAddress, 'pending');
   }
```

Comment:
Could this be better written as:

```suggestion
  getTransactionCount(walletAddress: string) => this.wallet.provider.getTransactionCount(walletAddress, 'pending');
```

Maybe?

Reply:
This definition won't compile as this is `class'.